components/script/dom/htmltableelement.rs

```diff
@@ -177,7 +178,7 @@ impl HTMLTableElementMethods for HTMLTableElement {
                 caption
             }
         };
-        Root::upcast(caption)
+        caption
```

Comment:
nit: This line can be removed altogether, and the let-binding removed, making this function contain only a match expression instead.